src/basicfuns.jl

```diff
@@ -11,15 +11,20 @@ julia> StatsFuns.xlogx(0)
 0.0
 ```
 """
-xlogx(x::Real) = x > zero(x) ? x * log(x) : zero(log(x))
+function xlogx(x)
```

Comment:
Are there any other functions where we allow any type? At least I'd use `::Number` here. Also the docstring above must be synced. Same below.

Reply:
Ok, `::Number` seems fine.